src/data/IFormState.ts

```diff
@@ -0,0 +1,24 @@
+// The FormStatus is to represent FormInput status
+export interface IFormState {
+    [id: string]: IFieldState;
+}
+
+export interface IFieldState {
+    error: string,
```

Comment:
I think `error: any` might be a better type so that complicated fields can store there errors as objects